# mocha-unhandled-rejections-example

It is passed, but We want to fail it...

```js
async function fail() {
    throw new Error("FAIL!!");
}

it("is unhandled rejection", () => {
    fail();
});
```

Fix the above:

```js
process.on("unhandledRejection", reason => {
    throw reason;
});

async function fail() {
    throw new Error("FAIL!!");
}

it("is unhandled rejection", () => {
    fail();
});
```

## Changelog

See [Releases page](https://github.com/azu/mocha-unhandled-rejections-example/releases).

## Running tests

Install devDependencies and Run `npm test`:

    npm test

## Contributing

Pull requests and stars are always welcome.

For bugs and feature requests, [please create an issue](https://github.com/azu/mocha-unhandled-rejections-example/issues).

1. Fork it!
2. Create your feature branch: `git checkout -b my-new-feature`
3. Commit your changes: `git commit -am 'Add some feature'`
4. Push to the branch: `git push origin my-new-feature`
5. Submit a pull request :D

## Author

- [github/azu](https://github.com/azu)
- [twitter/azu_re](https://twitter.com/azu_re)

## License

MIT Â© azu
